js/src/main/scala/BrowserCompiler.scala

```diff
@@ -187,62 +161,25 @@ object BrowserCompiler {
       compileResult.fold(compilationFailureJson, compilationSuccessJson)
   }
 
-  implicit object widgetCompilation2Json extends JsonWriter[WidgetCompilation] {
-    import BrowserCompiler.compiledWidgets2Json
-
-    def apply(widgetCompilation: WidgetCompilation): TortoiseJson =
-      widgetCompilation match {
-        case NotCompiled              => JsObject(fields())
-        case s: SourceCompilation     => Jsonify.writer[SourceCompilation, TortoiseJson](s)
-        case u: UpdateableCompilation => Jsonify.writer[UpdateableCompilation, TortoiseJson](u)
-        case p: PlotWidgetCompilation => Jsonify.writer[PlotWidgetCompilation, TortoiseJson](p)
-        case s: SliderCompilation     => Jsonify.writer[SliderCompilation, TortoiseJson](s)
-      }
-  }
-
   implicit def exception2Json(ex: Throwable): JsonWritable =
     JsonWriter.convert(ex)
 
   implicit def failure2Json(f: Failure): JsonWritable =
     JsonWriter.convert(f)
 
-  implicit def compileWidget2Json(compiledWidget: CompiledWidget): JsonWritable =
-    new JsonWritable {
-      def toJsonObj = JsObject(
-        widgetCompilation2Json(compiledWidget.widgetCompilation).asInstanceOf[JsObject].props ++
-        compiledWidget.widgetData.toJsonObj.asInstanceOf[JsObject].props)
-    }
-
   implicit def compilationResult2Json(modelCompilation: ValidationNel[Failure, ModelCompilation]): JsonWritable =
     JsonWriter.convert(modelCompilation)
 
-  case class CompiledWidget(widgetData: Widget, widgetCompilation: WidgetCompilation)
-
-  sealed trait WidgetCompilation
-  case object NotCompiled extends WidgetCompilation
-  case class SourceCompilation(
-    compiledSource: CompiledStringV) extends WidgetCompilation
-  case class UpdateableCompilation(
-    compiledSetupCode: CompiledStringV,
-    compiledUpdateCode: CompiledStringV) extends WidgetCompilation
-  case class PlotWidgetCompilation(
-    compiledSetupCode: CompiledStringV,
-    compiledUpdateCode: CompiledStringV,
-    compiledPens: Seq[CompiledWidget]) extends WidgetCompilation
-  case class SliderCompilation(compiledMin: CompiledStringV,
-    compiledMax: CompiledStringV,
-    compiledStep: CompiledStringV) extends WidgetCompilation
-
   case class ModelCompilation(
     result: String,
     code: String,
     info: String,
-    widgets: Seq[CompiledWidget],
+    widgets: List[JsObject],
```

Comment:
a `Seq` might be mutable (until https://issues.scala-lang.org/browse/SI-6570 gets fixed)